src/IW/Sync/Search.hs

```diff
@@ -0,0 +1,116 @@
+       ) 
+    => [Label] 
+    -> m [GitHub.Issue]
+fetchHaskellIssuesByLabels = liftGitHubSearchToApp . searchHaskellIssuesByLabels
+
+-- | Fetch all open issues with Haskell language.
+fetchAllHaskellIssues 
+    :: ( WithDb env m
+       , WithError m
+       , WithLog env m
+       ) 
+    => m [GitHub.Issue]
+fetchAllHaskellIssues = fetchHaskellIssuesByLabels []
+
+-- | Lift a github search action to work within our monad stack.
+liftGitHubSearchToApp 
+    :: forall a env m.
+       ( WithDb env m
+       , WithError m
+       , WithLog env m
+       ) 
+    => IO (Either GitHub.Error (SearchResult a)) 
+    -> m [a]
+liftGitHubSearchToApp githubSearch = liftIO githubSearch >>= \case
+    Left err -> throwError $ githubErrToAppErr err 
+    Right (SearchResult count vec) -> do
+        log Info $ "Fetching total of " <> show count <> " results..." 
+        pure $ V.toList vec
+
+-- | Search all repositories built with the Haskell language.
+searchHaskellRepos :: IO (Either GitHub.Error (SearchResult GitHub.Repo))
+searchHaskellRepos = searchRepos "language:haskell"
+
+-- | Construct a github search query from a list of labels.
+labelsToSearchQuery :: [Label] -> Text
+labelsToSearchQuery = foldMap (\Label{..} -> "label:\"" <> unLabel <> "\" ")
+
+-- | Search for all open Haskell issues with the corresponding labels.
+searchHaskellIssuesByLabels :: [Label] -> IO (Either GitHub.Error (SearchResult GitHub.Issue))
+searchHaskellIssuesByLabels labels = searchIssues $ "language:haskell is:open " <> labelsToSearchQuery labels
+
+-- | Convert a value of the @GitHub.Repo@ type to a value of our own @Repo@ type.
+githubRepoToRepo :: GitHub.Repo -> Repo
+githubRepoToRepo githubRepo = Repo{..}
+  where  
+    repoOwner      = RepoOwner . GitHub.untagName . GitHub.simpleOwnerLogin $ GitHub.repoOwner githubRepo
+    repoName       = RepoName  . GitHub.untagName $ GitHub.repoName githubRepo
+    repoDescr      = fromMaybe "" $ GitHub.repoDescription githubRepo
+    repoCategories = SqlArray []
+
+-- | Convert a value of the @GitHub.Issue@ type to a value of our own @Issue@ type.
+githubIssueToIssue :: GitHub.Issue -> Maybe Issue
+githubIssueToIssue githubIssue = do
+    issueRepoOwner <- fst <$> parseIssueUserData issueUrl
+    issueRepoName  <- snd <$> parseIssueUserData issueUrl
+    pure Issue{..}
+  where  
+    issueUrl    = GitHub.issueUrl githubIssue
+    issueNumber = GitHub.unIssueNumber $ GitHub.issueNumber githubIssue
+    issueTitle  = GitHub.issueTitle githubIssue 
+    issueBody   = fromMaybe "" $ GitHub.issueBody githubIssue
+    issueLabels = SqlArray . V.toList $ (\GitHub.IssueLabel{..} -> Label $ GitHub.untagName labelName) <$> GitHub.issueLabels githubIssue 
+
+-- | For parsing the @issueRepoOwner@ and @issueRepoName@ from the issue's URL.
+parseIssueUserData :: GitHub.URL -> Maybe (RepoOwner, RepoName)
+parseIssueUserData (URL url) =
+    ( T.stripPrefix "git@github.com:"     url
+  <|> T.stripPrefix "https://github.com/" url
```

Comment:
I recall you said that the URL for issue (that comes from GitHub) looks like this:

```
https://api.github.com/repos/<repoOwner>/<repoName>/issues/<number>
```

Should be in that case it be `T.stripPrefix "https://api.github.com/repos/"`?

Reply:
@chshersh Ooops yes that is correct!

Reply:
@chshersh And I don't have to worry about a `git@github.com:` prefix right?

Reply:
@rashadg1030 Right, you don't need to worry about alternative prefixes, we have only single prefix.